internal/services/storage/storage_account_resource_test.go

```diff
@@ -67,6 +67,70 @@ func TestAccStorageAccount_requiresImport(t *testing.T) {
 	})
 }
 
+func TestAccStorageAccount_recreateBasic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_account", "test")
+	r := StorageAccountResource{}
+
+	// fix location to test recreation
+	data.Locations.Primary = "westeurope"
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			// destroy the account and recreate it
+			Config:  r.recreate(data),
+			Destroy: true,
+		},
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStorageAccount_recreateWithProperties(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_account", "test")
+	r := StorageAccountResource{}
+
+	// fix location to test recreation
+	data.Locations.Primary = "westeurope"
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.recreate(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			// destroy the account and recreate it
+			Config:  r.recreate(data),
+			Destroy: true,
```

Comment:
this would destroy the resource group too, instead we can delete just the account via:

```suggestion
		{
			// destroy the account so that we can recreate it below
			Config:  r.template(data),
```